src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,167 @@
+                return;
+            }
+        }
+        //Converting datetime
+        if (command.equals("deadline") || command.equals("event")) {
+            try {
+                dateTime = LocalDateTime.parse(date + "T" + time);
+            } catch (DateTimeParseException e) {
+                if (command.equals("deadline")) {
+                    this.exception = new DukeException("Please add a date and time for your deadline in the following"
+                            + "format:\n'deadline <name> /by <YYYY-MM-DD> <HH:MM>' ! *quack*");
+                    return;
+                } else if (command.equals("event")) {
+                    this.exception = new DukeException("Please add a date and time for your event in the following"
+                            + "format:\n'event <name> /on <YYYY-MM-DD> <HH:MM>' ! *quack*");
+                    return;
+                }
+            }
+        }
+        this.taskType = command;
+        this.taskName = name;
+        this.taskPreposition = preposition;
+        this.taskDate = dateTime;
+    }
+
+    /**
+     * Executes the add command and adds the specified task and its details to Duke.
+     * If any exceptions were thrown during the parsing of user input and the initializiation of the command,
+     * they will be thrown here and the command will be terminated.
+     *
+     * @param tasks   The current tasks for the command to interact with.
+     * @param ui      The ui for the command to print output.
+     * @param storage The storage for the command to save and load tasks to an external file.
+     * @throws DukeException
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (this.exception != null) {
+            throw this.exception;
+        }
+
+        //Creating the new task
+        Task newTask = new Task("");
+        if (this.taskType.equals("todo")) {
+            newTask = new ToDoTask(this.taskName);
+        } else if (this.taskType.equals("deadline")) {
+            newTask = new DeadlineTask(this.taskName, this.taskPreposition, this.taskDate);
+        } else if (this.taskType.equals("event")) {
+            newTask = new EventTask(this.taskName, this.taskPreposition, this.taskDate);
+        } else {
+            throw new DukeException("Invalid command type");
+        }
+
+        //Adding the task
+        tasks.add(newTask);
+
+        //Output to update user
+        ui.print("Got it. I've added this task *quack*:");
+        ui.print(String.format("  %s", newTask.toString()));
+        ui.print(String.format(
+                "Now you have %d task%s in the list *quack*.",
+                tasks.getSize(), tasks.getSize() == 1 ? "" : "s"));
+
+        //Saving the task to the save file
+        storage.save(tasks);
+    }
+
+    /**
+     * Returns false as this is not an exit command.
+     *
+     * @return
```

Comment:
Shouldn't you say something here? I think just a `False.` would suffice

Reply:
agreed with @Rdac0 , otherwise it is ambiguous